azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/manager/AudioModeManager.kt

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.calling.presentation.manager
+
+import android.content.Context
+import android.media.AudioManager
+import android.media.AudioManager.MODE_IN_COMMUNICATION
+import android.media.AudioManager.MODE_NORMAL
+import com.azure.android.communication.ui.calling.redux.Store
+import com.azure.android.communication.ui.calling.redux.state.CallingStatus
+import com.azure.android.communication.ui.calling.redux.state.ReduxState
+import kotlinx.coroutines.flow.collect
+
+internal class AudioModeManager(
+    private val store: Store<ReduxState>,
+    context: Context,
+) {
+    private val audioManager by lazy { context.getSystemService(Context.AUDIO_SERVICE) as AudioManager }
+
+    suspend fun start() {
+        store.getStateFlow().collect {
+            if (audioManager.mode != MODE_IN_COMMUNICATION && it.callState.callingStatus == CallingStatus.CONNECTED) {
+                // to fix samsung device audio issue
+                // MODE_IN_COMMUNICATION is used to let the system know that the app is in a VOIP call
+                audioManager?.mode = MODE_IN_COMMUNICATION
+            }
```

Comment:
else {}
(assuming connected is the only state we want to be MODE_IN_COMMUNICATION)

Reply:
or you can one line it
audioManager.mode = it.callState.callingStatus == CallingStatus.CONNECTED?MODE_IN_COMMUNICATION:MODE_NORMAL;

Reply:
There are other call state as well. call connecting, disconnecting, resume etc. We only want MODE_IN_COMMUNICATION in connected more and only want to bring back to MODE_NORMAL when we go to hold. So that it releases the audio. Otherwise in API Level 29, It can not gain back if doesnt release volunterily.

Reply:
So writing `audioManager.mode = it.callState.callingStatus == CallingStatus.CONNECTED?MODE_IN_COMMUNICATION:MODE_NORMAL;` would make it more buggy as on other cases it would go to MODE_NORMAL which we dont want.